libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -689,15 +698,75 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
 
     // Initialize the pnpmfile if it doesn't exist
     if (!this._pnpmfileConfiguration) {
-      this._pnpmfileConfiguration = await PnpmfileConfiguration.initializeAsync(project.rushConfiguration, {
-        variant
-      });
+      this._pnpmfileConfiguration = await PnpmfileConfiguration.initializeAsync(
+        project.rushConfiguration,
+        subspaceName,
+        {
+          variant
+        }
+      );
+    }
+
+    let transformedPackageJson: IPackageJson = packageJson;
+
+    let subspacePnpmfile: IPnpmfile | undefined;
+    if (subspaceName) {
```

Comment:
@william2958 Why is this logic based on `subspaceName` -- isn't it basically checking whether the `--subspace` was passed to `rush update` or not?  

It seems that `pnpmfileGlobalFilename` should be checked for the default subspace as well. So shouldn't this `if` statement instead be testing whether the feature is enabled?